### PR TITLE
UEFI: Implement VM suspend/resume

### PR DESCRIPTION
**Pull request by edwintorok**
Continuation of https://github.com/xapi-project/xenopsd/pull/536

### CODE REVIEW

**Comment by lindig** [comment]
I think the `debug` messages should include the domid. The VM can be usually also inferred from the task in the logs but the message is easier to understand with the domid. For logging remember that varstored is a service per VM.

**Comment by edwintorok** [comment]
Thanks for the reviews so far, I've made some changes suggested during a review yesterday with @robhoes and @jonludlam . I think this is ready for a second round of review now.

File: xc/device_common.ml [hunk 1]
```diff
@@ -339,6 +339,9 @@ let demu_restore_path : (_, _, _) format = "/var/lib/xen/demu-resume.%d"
 let var_run_xen_path = "/var/run/xen"
 let qmp_libxl_path = (sprintf "%s/qmp-libxl-%d") var_run_xen_path
 let qmp_event_path = (sprintf "%s/qmp-event-%d") var_run_xen_path
+let efivars_init_path = (sprintf "%s/efi-vars-init-%d.dat") var_run_xen_path
```
Comment (rosslagerwall) [inline xc/device_common.ml]: This can be removed

File: xc/device.ml [hunk 2]
```diff
@@ -1921,17 +1974,26 @@ module Dm_Common = struct
...
-        let open Generic in
-        best_effort "killing vgpu"
-          (fun () -> really_kill vgpu_pid)
+    let stop_vgpu () = Vgpu.stop ~xs domid in
+    let stop_varstored () =
+      Varstored.stop ~xs domid;
+      List.iter (fun path ->
+          debug "deleting %s" path;
+          Unixext.unlink_safe path)
+        [ efivars_init_path domid
```
Comment (rosslagerwall) [inline xc/device.ml]: This can be removed

File: xc/device_common.mli [hunk 3]
```diff
@@ -98,6 +98,9 @@ val demu_restore_path: (int -> 'a, 'b, 'a) format
 val var_run_xen_path: string
 val qmp_libxl_path: int -> string
 val qmp_event_path: int -> string
+val efivars_init_path: int -> string
```
Comment (rosslagerwall) [inline xc/device_common.mli]: This can be removed

**Pull request merged**